### PR TITLE
Add tests for analysis protocol fields (#61)

### PR DESCRIPTION
**Pull request by danielsotirhos**

### CODE REVIEW

File: src/humancellatlas/data/metadata/api.py [hunk 1]
```diff
@@ -459,7 +459,27 @@ def __init__(self, json: JSON):
...
 class AnalysisProtocol(Protocol):
-    pass
+    describedBy: str
+    computational_method: str
+    provenance_document_id: str         # Entity self.document_id ?
+    provenance_submission_date: str
+    provenance_update_date: Optional[str]
+    protocol_core_protocol_id: str      # Protocol self.protocol_id ?
+    protocol_type_text: str
+
```
Comment (danielsotirhos) [inline src/humancellatlas/data/metadata/api.py]: Two of the AnalysisProtocol fields are already fetched by parent classes Protocol and Entity. Should these fields be removed from the AnalysisProtocol class to avoid duplication, or kept in AnalysisProtocol to make sure these fields exist for this class?

File: test/test.py [hunk 2]
```diff
@@ -485,6 +475,30 @@ def to_json(fqid):
 
         self.assertEqual({}, errors)
 
+    def test_sequencing_process_paired_end(self):
```
Comment (hannes-ucsc) [inline test/test.py]: Can you make the code move a separate commit?

Comment (hannes-ucsc) [reply test/test.py]: Could you elaborate the change from `'prod'` to `''`?

Comment (danielsotirhos) [reply test/test.py]: Yes, i'll make the code move a separate commit. I moved the function to group tests that don't use _test_bundle() below _test_bundle().

Comment (danielsotirhos) [reply test/test.py]: Regarding changing 'prod' to '': deployment = "prod" works for _load_bundle() but not if it needs to download the bundle because dss_client() needs it to be empty for prod.

Longer version: The "deployment" var is passed to _load_bundle().
In _load_bundle() the value of the var (or string "prod" if empty) is copied to a var "canning_directory".
If _load_bundle() needs ot download the bundle, dss_client() is called passing the "deployment" var.
dss_client() uses the "deployment" param to build the swagger url, requiring "deployment" to be empty if production is desired (otherwise you'd get https://dss.prod.data.humancellatlas.org...)

Comment (hannes-ucsc) [reply test/test.py]: I shouldn't have to modify the call to _load_bundle(). The same arguments should both work for initially downloading a bundle and canning it as well as loading an already canned bundle. Is there a bug lurking that you change is merely working around?

**Comment by hannes-ucsc** [comment]
Be sure to create a clean history. The deployment change should be one commit. The method move  should be another (even though I doubt it's necessary). The added test should be a third commit.

**Comment by danielsotirhos** [comment]
Rebased & cleaned history. One commit for the added test case, another commit to use the string 'prod' instead of a None type to indicate production deployments, and deleted the commit that moved the unrelated function "test_sequencing_process_paired_end()".

File: src/humancellatlas/data/metadata/helpers/dss.py [hunk 3]
```diff
@@ -92,14 +92,13 @@ def download_file(item):
     return bundle['version'], manifest, dict(metadata_files)
 
 
-def dss_client(deployment: Optional[str] = None) -> DSSClient:
+def dss_client(deployment: Optional[str] = 'prod') -> DSSClient:
```
Comment (hannes-ucsc) [inline src/humancellatlas/data/metadata/helpers/dss.py]: `Optional[T]` means `Union[None,T]`

We shouldn't permit `None` …

File: src/humancellatlas/data/metadata/helpers/dss.py [hunk 4]
```diff
@@ -92,14 +92,13 @@ def download_file(item):
...
     """
     Return a DSS client to DSS production or the specified DSS deployment.
 
-    :param deployment: The name of a DSS deployment like `dev`, `integration` or `staging`. If None, the production
-                       deployment (`prod`) will be used.
+    :param deployment: The name of a DSS deployment like `dev`, `integration` or `staging`.
     """
-    deployment = deployment + "." if deployment else ""
+    deployment = deployment + "." if deployment and deployment != 'prod' else ""
```
Comment (hannes-ucsc) [inline src/humancellatlas/data/metadata/helpers/dss.py]: … and we shouldn't accept False-ish values.

```suggestion
    deployment = "" if deployment == 'prod' else deployment + "." 
```

Also like to avoid negation as much as possible.

File: test/test.py [hunk 5]
```diff
@@ -143,18 +144,17 @@ def _canned_bundle(self, directory, uuid, version):
         else:
             return None, None
 
-    def _load_bundle(self, uuid, version, replica='aws', deployment=None):
+    def _load_bundle(self, uuid, version, replica='aws', deployment='prod'):
```
Comment (hannes-ucsc) [inline test/test.py]:
```suggestion
   
    prod = 'prod'

    def _load_bundle(self, uuid, version, replica='aws', deployment=prod):
```

**Comment by hannes-ucsc** [comment]
Commit titles appear to be the same but they should be different.

File: src/humancellatlas/data/metadata/helpers/dss.py (hunk 4 above)
Comment (danielsotirhos) [reply src/humancellatlas/data/metadata/helpers/dss.py]: Easier to read this way, agreed.

File: test/test.py (hunk 5 above)
Comment (danielsotirhos) [reply test/test.py]: I am unsure of the benefit of this suggested change.
